pkg/vpcmodel/semanticDiffSubnets.go

```diff
@@ -0,0 +1,212 @@
+package vpcmodel
+
+import (
+	"fmt"
+
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/common"
+)
+
+// ToDo: go over structs specifically * and lack of
+
+type DiffType = int
+
+const (
+	NoDiff DiffType = iota
+	MissingSrcEP
+	MissingDstEP
+	MissingSrcDstEP
+	MissingConnection
+	ChangedConnection
+)
+
+type connectionDiff struct {
+	*common.ConnectionSet
+	diff DiffType
+}
+
+type SubnetsDiff map[EndpointElem]map[EndpointElem]*connectionDiff
+
+type ConfigsForDiff struct {
+	config1 *CloudConfig
+	config2 *CloudConfig
+}
+
+type SubnetConfigConnectivity struct {
+	config             *CloudConfig
+	subnetConnectivity SubnetConnectivityMap
+}
+
+type diffBetweenSubnets struct {
+	subnet1Subtract2 SubnetsDiff
+	subnet2Subtract1 SubnetsDiff
+
+	GroupedSubnet1Minus2 *GroupConnLines
+	GroupedSubnet1Minus1 *GroupConnLines
+}
+
+func (configs ConfigsForDiff) GetSubnetsDiff(grouping bool) (*diffBetweenSubnets, error) {
+	// 1. compute connectivity for each of the subnets
+	subnetsConn1, err := configs.config1.GetSubnetsConnectivity(true, grouping)
```

Comment:
Actually the grouping is not relevant when calling GetSubnetsConnectivity , since the grouping of the diffs will be done separately, in any stage. Fixed.